Wrappers/Python/cil/utilities/display.py

```diff
@@ -92,6 +98,288 @@ def save(self,filename, **kwargs):
         except:
             print("Unable to save image")
 
+
+class show1D(show_base):
+    """
+    This creates and displays 1D plots of pixel values by slicing
+    multi-dimensional data.
+
+    The behaviour is as follows: if provided multiple datasets and a single
+    slice set (see first example below), one line plot will be generated
+    per dataset; if provided a single dataset and multiple sets of slices
+    (see second example below), one line plot will be generated per slice
+    set;  if provided multiple datasets and multiple slice sets, the
+    :math:`i`-th set of slices will apply to the :math:`i`-th dataset, with
+    a line plot generated in each case.
+
+    Parameters
+    ----------
+    data : DataContainer, list of DataContainer, tuple of DataContainer
+        Multi-dimensional data to be reduced to 1D.
+    line_coords : list of tuple or list of list of tuple, default=None
+        A list, or nested list, of (dimension, coordinate) pairs for
+        slicing `data` (default is None, which is only valid when 1D
+        data is passed)
+    label : str, list of str, default=None
+        Label(s) to use in the plot's legend
+    title : str, default None
+        A title for the plot
+    color : str, list of str, default=None
+        Color(s) for each line plot
+    ls : {"-","--","-.",":"}, dict of {"-","--","-.",":"}, default=None
+        Linestyle(s) for each line plot
+    axis_labels : tuple of str, list of str, default=('Pixel','Pixel value')
+        Axis labels in the form (x_axis_label,y_axis_label)
+    num_cols : int, default=3
+        The number of columns in the grid of subplots produced in the case
+        of multiple plots
+    size : tuple, default=(8,6)
+        The size of the figure
+    force : bool, default=True
+        Passed to `get_slice`
+
+    Attributes
+    ----------
+    figure : matplotlib.figure.Figure
+
+    Examples
+    --------
+
+    This example creates two 2D datasets (images), and uses the provided
+    slicing information to generate two plots on the same axis,
+    corresponding to the two datasets.
+
+    >>> from cil.utilities.display import show1D
+    >>> from cil.utilities.dataexample import PEPPERS
+    >>> data = PEPPERS.get()
+    >>> data_channel0 = data.get_slice(channel=0)
+    >>> data_channel1 = data.get_slice(channel=1)
+    >>> show1D([data_channel0, data_channel1], line_coords=[('horizontal_x', 256)],
+    ...        label=['Channel 0', 'Channel 1'])
```

Comment:
```suggestion
    >>> show1D([data_channel0, data_channel1], line_coords=[('horizontal_x', 256)],
    ...        label=['Channel 0', 'Channel 1'], ls=["--", "-"])
```
Would just like to add setting the line style to the example